raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -840,7 +840,7 @@ def events_for_onchain_secretreveal_if_closed(
         transfers_pair: List[MediationPairState],
         secret: Secret,
         secrethash: SecretHash,
-) -> List[ContractSendSecretReveal]:
+) -> List[Event]:
```

Comment:
But this change makes the type more generic. Why introduce it? We are 100% sure that the List of events here will only contain `ContractSendSecretReveal`. I also like seeing the type annotations when reading the code to understand what's happening. To me `List[ContractSendSecretReveal]` is a lot more helpful than `List[Event]`. Please find another way to appease mypy. It should work for us and not against us.

Reply:
`List[ContractSendSecretReveal]` is not more useful than `List[Event]`. `List[Event]` can take `.append(object_of_another_event)` but `List[ContractSendSecretReveal]` cannot take `.append(object_of_another_event)`. See "List[subclass] is incompatible with List[superclass]" https://github.com/python/mypy/issues/2984 It's because you can update a List in-place.

The alternative is to use a Sequence instead of List.  So we'll be using `(event1, event2, event3)`.  Sequences are not updatable, `Sequence[ContractSendSecretReveal]` is more useful than `Sequence[Event]`.

@LefterisJP Shall we go with `Sequence[ContractSendSecretReveal]` here?

Reply:
> List[ContractSendSecretReveal] is not more useful than List[Event]

I disagree. Type annotations also serve the purpose of documentation for humans, not only mypy.

> but List[ContractSendSecretReveal] cannot take .append(object_of_another_event)

Exactly. It should not take such an argument for as long as we are in this function. This function can only append `ContractSendReveal` to the events list. I wonder if there is any way to show this in mypy.

> @LefterisJP Shall we go with Sequence[ContractSendSecretReveal] here?

I have not tried it. Will it work? Sequence is immutable and we modify the events list.

Reply:
> Will it work?

Yes but with a runtime cost.
* `Sequence[ContractSendSecretReveal]` is a subtype of `Sequence[Event]`.  So it's safe to concatenate these sequences. `List[ContractSendSecretReveal]` is not a subtype of `List[Event]`.
* At runtime `Sequence[]` will be slower. Every time the event list is modified, a completely new tuple needs to be created.

Reply:
Hm then no. We are not gonna change the runtime to appease mypy.

This needs more thinking.

Reply:
I agree. We won't introduce runtime burden for mypy.

Reply:
Oh but `Sequence` can be a `Tuple` or `List` or something else.  So I found a solution without a runtime cost.

Reply:
yes this is what the mypy thread you linked seems to indicate.